### PR TITLE
Make option names derived from ref pretty

### PR DESCRIPTION
**Pull request by rbruggem**
The ref name, which is camelcase, is split by word then sentence-cased.

Before:
Screenshot 2023-06-21 at 16 54 31

After:
Screenshot 2023-06-21 at 16 54 14

In some instances this change is not going to yield the correct outcome.
Eg. `IATACode` -> `I a t a code`

I will provide a way to handle special cases in upcoming PRs.

### CODE REVIEW

**Comment by rbruggem** [comment]
> I will provide a way to handle special cases in upcoming PRs.

https://github.com/lune-climate/lune-backend/pull/3585/files

**Review by raguiar9080 — APPROVED** [review]
I like it 👍 



Random thought: This PR is #358 and references #3585 and I was a bit confused at first since I thought they were the same 😄 Cool coincidence

File: src/components/Dereferencer.tsx [hunk 1]
```diff
@@ -15,10 +16,13 @@ export default function Dereferencer(element: any, name?: string): any {
         const schemaElement = {
             ...tokenized.slice(1).reduce((acc, current) => acc[current], schema),
         }
+        const nameFromToken = tokenized[3]
+            ? to.sentence(tokenized[3].replace(/([A-Z])/g, ' $1'))
```
Comment (jstasiak) [inline src/components/Dereferencer.tsx:20]: Two nitpicks:

* Can we have this in a separate function and have some unit tests for it?
* Is it really worth pulling a bunch of dependencies to achieve this result?

**Review by jstasiak — APPROVED** [review]

File: src/components/Dereferencer.tsx (hunk 1 above)
Comment (rbruggem) [reply src/components/Dereferencer.tsx:20]: > Two nitpicks:
> 
> * Can we have this in a separate function and have some unit tests for it?
> * Is it really worth pulling a bunch of dependencies to achieve this result?

@jstasiak 
https://github.com/lune-climate/lune-docs/pull/361

Comment (jstasiak) [reply src/components/Dereferencer.tsx:20]: Oh, sorry – I actually thought we had some tests set up here, otherwise I'd not have mentioned that

**Pull request merged**